### PR TITLE
feat: settings page

### PR DESCRIPTION
**Pull request by yanzhihong23**

### CODE REVIEW

File: packages/dashboard/src/pages/Settings/regexp.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export const passwordRegexp = /^(?=.{8,20}$)(?=.*[A-Z])(?=.*[a-z])(?=.*[0-9])(?=.*[^A-Za-z0-9]).*/
+
+export const emailRegexp = /^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$/
```
Comment (Jack-Works) [inline packages/dashboard/src/pages/Settings/regexp.ts:3]: [Stop Validating Email Addresses With Regex](https://davidcel.is/posts/stop-validating-email-addresses-with-regex/). If you need to check email address, please use `<input type="email" />`

File: packages/dashboard/src/pages/Settings/regexp.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+export const phoneRegexp = /(\+?( |-|\.)?\d{1,2}( |-|\.)?)?(\(?\d{3}\)?|\d{3})( |-|\.)?(\d{3}( |-|\.)?\d{4})/
```
Comment (Jack-Works) [inline packages/dashboard/src/pages/Settings/regexp.ts:5]: Also, don't use regexp to match phone numbers. Do you sure your phone number can cover different phone numbers from different countries? If you really need it, please use `<input type="tel" />` or https://github.com/google/libphonenumber

File: packages/dashboard/src/pages/Settings/api.ts [hunk 3]
```diff
@@ -13,3 +13,37 @@ export const [useAncientPostsCompatibilityMode] = createGlobalState(
     Services.Settings.getAncientPostsCompatibiltyMode,
     (x) => Messages.events.disableOpenNewTabInBackgroundSettings.on(x),
 )
+
+const API_HOST = 'https://vaalh28dbi.execute-api.ap-east-1.amazonaws.com'
```
Comment (septs) [inline packages/dashboard/src/pages/Settings/api.ts:17]: this api gateway instance, are we in owner?

CC @nykma and @yisiliu

Comment (nykma) [reply packages/dashboard/src/pages/Settings/api.ts:17]: Yes. This is the cloud backup server (staging) entrypoint

Comment (septs) [reply packages/dashboard/src/pages/Settings/api.ts:17]: plz add a comment (this is temporary api endpoint)

File: packages/dashboard/src/components/CountdownButton/index.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    const { duration = 60, children, onClick, ...others } = props
+    const [countdown, setCountdown] = useState(0)
+    const handleClick = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+        setCountdown(duration)
+        onClick && onClick(event)
+    }
+
+    useEffect(() => {
+        if (countdown) {
+            setTimeout(() => {
```
Comment (guanbinrui) [inline packages/dashboard/src/components/CountdownButton/index.tsx:18]: It needs to clean the timer.

File: packages/dashboard/src/pages/Settings/components/dialogs/LogoutDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        onClose()
+    }
+    return (
+        <MaskDialog title={t.settings_log_out_title()} maxWidth="xs" open={open} onClose={onClose}>
+            <DialogContent sx={{ padding: '16px 44px 64px' }}>
+                <Typography sx={{ fontSize: '14px' }}>{t.settings_log_out_tips()}</Typography>
+            </DialogContent>
+            <DialogActions>
+                <StyledButton onClick={onClose} color="secondary">
+                    Cancel
```
Comment (guanbinrui) [inline packages/dashboard/src/pages/Settings/components/dialogs/LogoutDialog.tsx:39]: i18n

File: packages/dashboard/src/pages/Settings/components/dialogs/SettingEmailDialog.tsx [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+import { useDashboardI18N } from '../../../../locales'
+import CountdownButton from '../../../../components/CountdownButton'
+import { sendCode, verifyCode } from '../../api'
+import { emailRegexp } from '../../regexp'
+
+const useStyles = makeStyles({
+    container: {
+        minHeight: '200px',
+        paddingLeft: '20px',
+        paddingRight: '20px',
```
Comment (guanbinrui) [inline packages/dashboard/src/pages/Settings/components/dialogs/SettingEmailDialog.tsx:14]:
```suggestion
        minHeight: 200,
        paddingLeft: 20,
        paddingRight: 20,
```

File: packages/dashboard/src/pages/Settings/components/SettingItem.tsx [hunk 7]
```diff
@@ -28,14 +29,17 @@ const TextWrapper = styled(ListItemText)(({ theme }) => ({
...
 }))
 
 export default function SettingItem(props: SettingItemProps) {
-    const { title, desc, icon } = props
+    const { title, desc, icon, error = false } = props
     return (
         <ListItem sx={{ paddingLeft: 0 }}>
             <IconWarpper>{icon}</IconWarpper>
-            <TextWrapper primary={title} secondary={desc} />
+            <TextWrapper primary={title} secondary={desc} className={error ? 'error' : ''} />
```
Comment (septs) [inline packages/dashboard/src/pages/Settings/components/SettingItem.tsx:42]:
```suggestion
            <TextWrapper primary={title} secondary={desc} className={error ? 'error' : undefined} />
```

**Review by septs — COMMENTED** [review]
plz resolve all eslint errors

File: packages/dashboard/src/pages/Settings/components/dialogs/CloudBackupVerifyDialog.tsx [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+    const t = useDashboardI18N()
+    const { user } = useContext(UserContext)
+    const [mode, setMode] = useState((user.email ?? user.phone) || '')
+    const [code, setCode] = useState('')
+    const [invalidCode, setInvalidCode] = useState(false)
+
+    const params = useMemo(() => {
+        return {
+            account: mode,
+            type: (/@/.test(mode) ? 'email' : 'phone') as AccountValidationType,
```
Comment (guanbinrui) [inline packages/dashboard/src/pages/Settings/components/dialogs/CloudBackupVerifyDialog.tsx:32]:
```suggestion
            type: (mode.includes('@') ? 'email' : 'phone') as AccountValidationType,
```

File: packages/icons/general/CloudBackup.tsx [hunk 9]
```diff
@@ -0,0 +1,49 @@
+import { createIcon } from '../utils'
+export const CloudBackupIcon = createIcon(
+    'CloudBackupIcon',
+    <>
```
Comment (guanbinrui) [inline packages/icons/general/CloudBackup.tsx:4]: Use `<g />` to wrap SVG compoent.

**Review by guanbinrui — APPROVED** [review]

File: packages/dashboard/src/pages/Settings/components/dialogs/BackupDialog.tsx [hunk 10]
```diff
@@ -1,56 +1,151 @@
...
     open: boolean
     onClose(): void
 }
 
-export default function BackupDialog({ open, onClose }: BackupDialogProps) {
+export default function BackupDialog({ local = true, params, open, onClose }: BackupDialogProps) {
+    const snackbar = useSnackbar()
     const t = useDashboardI18N()
-    const [checked, setChecked] = useState(true)
+    const [backupPassword, setBackupPassword] = useState('')
```
Comment (septs) [inline packages/dashboard/src/pages/Settings/components/dialogs/BackupDialog.tsx:24]: plz use react-hook-form build the pr form

**Review by septs — CHANGES_REQUESTED** [review]

File: packages/dashboard/src/pages/Settings/components/dialogs/BackupDialog.tsx (hunk 10 above)
Comment (yanzhihong23) [reply packages/dashboard/src/pages/Settings/components/dialogs/BackupDialog.tsx:24]: I think this should be optional. We can use it in a complex form.

**Pull request merged**